src/ngraph/frontend/onnx_import/op/matmul.cpp

```diff
@@ -27,7 +27,18 @@ namespace ngraph
             {
                 NodeVector matmul(const Node& node)
                 {
-                    auto factory = matmul::MatmulFactory(node);
+                    auto factory = builder::MatmulFactory(node.get_ng_inputs());
+                    std::size_t left_rank{factory.get_left()->get_shape().size()};
+                    std::size_t right_rank{factory.get_right()->get_shape().size()};
+
+                    if (left_rank == 0 || right_rank == 0)
+                    {
+                        NGRAPH_WARN
+                            << (node) << " "
+                            << "ONNX standard doesn't allow scalar operands, however nGraph "
```

Comment:
I'd suggest throwing an exception here, rather than just logging it. (Similar comment where an analogous check occurs in other files.)

Reply:
Our intention is to follow the robustness principle and "be liberal in what you accept from others".

Reply:
Not sure I agree that's the right principle to follow here, but I can agree to disagree.